import api from "../api";
import { AxiosError } from 'axios';

export const buscarSumario = async (ano?: string): Promise<any> => {
    try {
        const params = ano ? { ano } : {};
        console.log('Par칙metros enviados para a API:', params); // Log dos par칙metros enviados
        const resposta = await api.get('/projeto/sumario', { params });
        console.log('Resposta da API:', resposta); // Log da resposta da API
        if (resposta.status === 200) {
            return { status: resposta.status, data: resposta.data };
        } else {
            return { status: resposta.status, message: resposta.data };
        }
    } catch (error) {
        const axiosError = error as AxiosError<{ message?: string }>;
        const errorMessage = axiosError.response?.data?.message || 'Erro ao carregar o sum치rio. Por favor, tente novamente mais tarde.';
        console.error('Erro ao carregar o sum치rio:', errorMessage); // Log do erro
        throw new Error(errorMessage);
    }
};